src/Omega_h_simModel2d.cpp

```diff
@@ -0,0 +1,466 @@
+#include <SimModel.h>
+#include <SimUtil.h>
+#include "Omega_h_model2d.hpp"
+#include "Omega_h_profile.hpp"
+#include "Omega_h_map.hpp"  // get_degrees
+#include "Omega_h_adj.hpp"  // invert_adj
+#include "Omega_h_array_ops.hpp" // operator==(LOs,LOs)
+#include <map>
+#include <algorithm> //std::fill
+#include <numeric> //std::exclusive_scan
+
+namespace Omega_h {
+
+HostWrite<LO> createAndInitArray(size_t size, const LO init=0) {
+  auto array = HostWrite<LO>(size);
+  std::fill(array.data(), array.data()+array.size(), init);
+  return array;
+}
+
```

Comment:
these constructors can also be used, cant they ? https://github.com/SCOREC/omega_h/blob/ef317f8f5c52f2d2eb74eea5a9ecf8665088ab98/src/Omega_h_array.hpp#L164

Reply:
https://github.com/SCOREC/omega_h/blob/ef317f8f5c52f2d2eb74eea5a9ecf8665088ab98/src/Omega_h_array.hpp#L162-L166
For `HostWrite(...)`, the closest ctor I see is the one that takes an initializer list, but I don't think that will support setting all the values in the array when its length is set at runtime.  I also thought there would have been one that initialized all the values but I didn't find it.
Interestingly, there are `Write()` and `Read()` ctors that support it:
https://github.com/SCOREC/omega_h/blob/ef317f8f5c52f2d2eb74eea5a9ecf8665088ab98/src/Omega_h_array.hpp#L64
https://github.com/SCOREC/omega_h/blob/ef317f8f5c52f2d2eb74eea5a9ecf8665088ab98/src/Omega_h_array.hpp#L95

Reply:
for this one 
HostWrite(LO size_in, T offset, T stride, std::string const& name = "");

Reply:
Yeah, good call.  I misunderstood what that ctor did.  Thanks.

For the record, it creates a HostWrite from a call to the Write(...) ctor with the same signature.  
https://github.com/SCOREC/omega_h/blob/ef317f8f5c52f2d2eb74eea5a9ecf8665088ab98/src/Omega_h_array.cpp#L223-L226
That in turn calls `fill_linear(...)`.
https://github.com/SCOREC/omega_h/blob/ef317f8f5c52f2d2eb74eea5a9ecf8665088ab98/src/Omega_h_array.cpp#L67-L73